python/ct/proto/client_v2.proto

```diff
@@ -0,0 +1,180 @@
+syntax = "proto2";
+
+package ct;
+
+import "ct/proto/client.proto";
+import "ct/proto/tls_options.proto";
+
+// Actually X509ChainEntry in RFC6962-bis
+message X509ChainEntryV2 {
+  // leaf_certificate, opaque (ASN.1Cert) <1..2^24-1>
+  optional bytes leaf_certificate = 1 [ (tls_opts).min_length = 1,
+                                        (tls_opts).max_length = 0xffffff ];
+  // certificate_chain, repeated opaque (ASN.1Cert) <0..2^24-1>
+  // A chain from the leaf to a trusted root, excluding the leaf, including
+  // the trusted root. May be empty if the submitted leaf_certificate is a
+  // trusted root.
+  repeated bytes certificate_chain = 2 [ (tls_opts).min_length = 0,
+                                         (tls_opts).max_length = 0xffffff,
+                                         (tls_opts).max_total_length = 0xffffff ];
+}
+
+message PrecertChainEntryV2 {
+  // pre_certificate, opaque (CMSPrecert) <1..2^24-1>
+  optional bytes pre_certificate = 1 [
+    (tls_opts).min_length = 1,
+    (tls_opts).max_length = 0xffffff ];
+
+  // precertificate_chain, repeated opaque (ASN.1Cert) <1..2^24-1>
+  // A chain from the precert to a trusted root, excluding the precert itself.
+  // Must have at least one certificate.
+  repeated bytes precertificate_chain = 2 [
```

Comment:
Not sure I understand why: In 6962-bis, the PrecertChainEntryV2 contains precertificate_chain and pre_certificate fields.
(are you suggesting we change that in 6962-bis to pre_certificate_chain?)